### PR TITLE
[Map] Polygons for Farm's Regions

### PR DESCRIPTION
**Pull request by GuessingMan**
## Summary
- Added Polygons to the MapView using latitude and longitude data that was taken from manual mappings done in MapBox
- Created a getCoordinates function that maps the information in MapBox into Polygons
- MapBox data also includes info on the name of a region and type of land that region is on
- Created a Feature type that represents a dictionary from MapBox data

## Test Plan
- plan to review
### Next Steps
- We can look to use polygons to create visible regions on the map
- We can add more polygons to encompass every region in Alemany Farm
- We can look to name Polygons so that they can be identified once tapped

### Relevant Links
Polygon Info:
https://github.com/react-native-maps/react-native-maps/blob/master/docs/polygon.md

Mapping Mapbox JSON:
https://pretagteam.com/question/how-to-draw-polygon-on-react-native-maps-with-many-pointslatitude-and-longitude

Polygon Youtube Tutorial: (Skip to 14:00 min)
https://www.youtube.com/watch?v=4qq0GQPkfjI

## Screenshots and Tests
unnamed

- if your feature is visual, attach before/after screenshots and ensure that tests still pass
CC: @mylesdomingo

### CODE REVIEW

File: src/screens/HomeScreen.tsx [hunk 1]
```diff
@@ -37,6 +65,44 @@ export default function HomeScreen() {
...
+              latitude: 37.732937,
+              longitude: -122.42002,
+            },
+          ]}
+          fillColor="rgba(0,0,255,0.5)"
+          strokeColor="rgba(0,0,255,0.5)"
+          strokeWidth={2}
+        >
+          <Callout>
+            <Text> Fruit Ally </Text>
```
Comment (mylesdomingo) [inline src/screens/HomeScreen.tsx:100]: whats the callout / text for?

File: src/screens/HomeScreen.tsx [hunk 2]
```diff
@@ -15,12 +17,38 @@ const styles = StyleSheet.create({
...
+  }
+};
+
 export default function HomeScreen() {
   const [trees, setTrees] = useState<Tree[]>([]);
 
   const isValidLocation = (tree: Tree) =>
     tree.location && tree.location.latitude && tree.location.longitude;
 
+  const getCoordinates = (mapboxJSON: Dictionary) => {
```
Comment (mylesdomingo) [inline src/screens/HomeScreen.tsx:32]: i wouldn't use the Dictionary type here it's not really type safe... do create a type safeguard

**Review by mylesdomingo — CHANGES_REQUESTED** [review]

File: src/screens/HomeScreen.tsx [hunk 3]
```diff
@@ -1,11 +1,13 @@
+/* eslint-disable no-restricted-syntax */
```
[the remark this answers is not in the document]
Comment (GuessingMan) [reply src/screens/HomeScreen.tsx:1]: i'm not sure why it was there. i deleted it

File: src/screens/HomeScreen.tsx [hunk 4]
```diff
@@ -37,6 +65,44 @@ export default function HomeScreen() {
   return (
     <ViewContainer>
       <MapView style={styles.map} region={DEFAULT_LOCATION}>
+        <Polygon
+          coordinates={[
+            {
+              latitude: 37.732937,
```
[the remark this answers is not in the document]
Comment (GuessingMan) [reply src/screens/HomeScreen.tsx:71]: this was a separate experimental polygon i was using, it's a copy of one of the ones already in a separate file. it's related to the Callout below

File: src/screens/HomeScreen.tsx (hunk 1 above)
Comment (GuessingMan) [reply src/screens/HomeScreen.tsx:100]: I saw in a video where they used <Callout> then <Text> to create labels for polygons. When I tried it here it did not work, should I delete this completely? It was more of an experimental thing and I know you said that PRs should be complete and not have extra stuff

File: src/screens/HomeScreen.tsx [hunk 5]
```diff
@@ -21,6 +22,26 @@ export default function HomeScreen() {
...
+    return features.map((feature: Feature) => {
+      const coordinates = feature.geometry.coordinates.map(
+        (coord: number[]) => ({
+          latitude: coord[1],
+          longitude: coord[0],
+        }),
+      );
+      return (
+        <Polygon
+          coordinates={coordinates}
```
Comment (mylesdomingo) [inline src/screens/HomeScreen.tsx:36]: can you add a unique key prop here? there's a warning. maybe the id provided in mapBox

**Review by mylesdomingo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**